pyatlan/client/workflow.py

```diff
@@ -0,0 +1,189 @@
+    WorkflowSearchResponse,
+    WorkflowSearchResult,
+    WorkflowSearchResultDetail,
+)
+
+MONITOR_SLEEP_SECONDS = 5
+
+
+@runtime_checkable
+class ApiCaller(Protocol):
+    def _call_api(
+        self, api, query_params=None, request_obj=None, exclude_unset: bool = True
+    ):
+        pass
+
+
+class WorkflowClient:
+    """
+    This class can be used to retrieve information and re_run workflows. This class does not need to be instantiated
+    directly by can be obtained through the workflow property of AtlanClient.
+    """
+
+    def __init__(self, client: ApiCaller):
+        if not isinstance(client, ApiCaller):
+            raise ErrorCode.INVALID_PARAMETER_TYPE.exception_with_parameters(
+                "client", "ApiCaller"
+            )
+        self._client = client
+
+    def find_by_type(
+        self, prefix: WorkflowPackage, max_results: int = 10
+    ) -> list[WorkflowSearchResult]:
+        """
+        Find workflows based on their type (prefix). Note: Only workflows that have been run will be found.
+
+        :param prefix: name of the specific workflow re_run to find (for example ATLAN_CONNECTION_DELETE)
+        :param max_results: the maximum number of results to retrieve
+        :returns: the list of workflows of the provided type, with the most-recently created first
+        """
+        if not isinstance(prefix, WorkflowPackage):
+            raise ErrorCode.INVALID_PARAMETER_TYPE.exception_with_parameters(
+                "prefix", "WorkflowPackage"
+            )
+        query = Bool(
+            filter=[
+                NestedQuery(
+                    query=Prefix(field="metadata.name.keyword", value=prefix.value),
+                    path="metadata",
+                )
+            ]
+        )
+        request = WorkflowSearchRequest(query=query, size=max_results)
+        raw_json = self._client._call_api(
+            WORKFLOW_INDEX_SEARCH,
+            request_obj=request,
+        )
+        response = WorkflowSearchResponse(**raw_json)
+        return response.hits.hits or []
+
+    def _find_latest_run(self, workflow_name: str) -> Optional[WorkflowSearchResult]:
+        query = Bool(
+            filter=[
+                NestedQuery(
+                    query=Term(
+                        field="metadata.name.keyword",
+                        value=workflow_name,
+                    ),
+                    path="metadata",
+                )
+            ]
+        )
+        response = self._find_run(query)
+        return results[0] if (results := response.hits.hits) else None
+
+    def _find_run(self, query: Query, size=1) -> WorkflowSearchResponse:
+        request = WorkflowSearchRequest(query=query, size=size)
+        raw_json = self._client._call_api(
+            WORKFLOW_INDEX_RUN_SEARCH,
+            request_obj=request,
+        )
+        return WorkflowSearchResponse(**raw_json)
+
+    @overload
+    def re_run(self, workflow: WorkflowPackage) -> WorkflowRunResponse:
```

Comment:
I'd suggest we use `rerun` rather than `re_run` — the former's a perfectly valid verb: https://dictionary.cambridge.org/dictionary/english/rerun